cmd/dep/main.go

```diff
@@ -55,13 +55,44 @@ func main() {
 		fmt.Fprintln(os.Stderr)
 	}
 
-	if len(os.Args) <= 1 || len(os.Args) == 2 && (strings.Contains(strings.ToLower(os.Args[1]), "help") || strings.ToLower(os.Args[1]) == "-h") {
+	// parseArgs determines the name of the dep command and wether the user asked for
+	// help to be printed.
+	parseArgs := func(args []string) (cmdName string, printCmdUsage bool, exit bool) {
+		isHelpArg := func() bool {
+			if strings.Contains(strings.ToLower(args[1]), "help") ||
+				strings.ToLower(args[1]) == "-h" {
+				return true
+			}
+			return false
+		}
+
+		switch len(args) {
+		case 0, 1:
+			exit = true
+		case 2:
+			if isHelpArg() {
+				exit = true
+			}
+			cmdName = args[1]
+		default:
+			if isHelpArg() {
+				cmdName = args[2]
+				printCmdUsage = true
+			} else {
+				cmdName = args[1]
+			}
+		}
+		return cmdName, printCmdUsage, exit
+	}
+
+	cmdName, printCommandHelp, exit := parseArgs(os.Args)
```

Comment:
`cmdName` should be used below instead of `os.Args[1]` below in the `no such command` error